python-spec/src/somacore/scene.py

```diff
@@ -0,0 +1,270 @@
+        the level ``0`` image.
+
+        Args:
+            key: The name of the multiscale image.
+            transform: The coordinate transformation from the scene to the reference
+                level of the multiscale image.
+            subcollection: The name, or sequence of names, of the subcollection the
+                image is stored in. Defaults to ``'img'``.
+            coordinate_space: Optional coordinate space for the image. This will
+                replace the existing coordinate space of the multiscale image.
+
+        Returns:
+            The registered multiscale image in write mode.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def register_point_cloud(
+        self,
+        key: str,
+        transform: coordinates.CoordinateTransform,
+        *,
+        subcollection: Union[str, Sequence[str]] = "obsl",
+        coordinate_space: Optional[coordinates.CoordinateSpace] = None,
+    ) -> _PointCloud:
+        """Adds the coordinate transform for the scene coordinate space to
+        a point cloud stored in the scene.
+
+        If the subcollection the point cloud is inside of is more than one
+        layer deep, the input should be provided as a sequence of names. For example,
+        to register a point named `transcripts` in the `var/RNA`
+        collection::
+
+            scene.register_point_cloud(
+                'transcripts', transform, subcollection=['var', 'RNA'],
+            )
+
+        Args:
+            key: The name of the point cloud.
+            transform: The coordinate transformation from the scene to the point cloud.
+            subcollection: The name, or sequence of names, of the subcollection the
+                point cloud is stored in. Defaults to ``'obsl'``.
+            coordinate_space: Optional coordinate space for the point cloud. This will
+                replace the existing coordinate space of the point cloud. Defaults to
+                ``None``.
+
+        Returns:
+            The registered point cloud in write mode.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_transformation_to_geometry_dataframe(
+        self, key: str, *, subcollection: Union[str, Sequence[str]] = "obsl"
+    ):
+        """Returns the coordinate transformation from the scene to a requested
+        geometery dataframe.
+
+        Args:
+            key: The name of the geometry dataframe.
+            subcollection: The name, or sequence of names, of the subcollection the
+                dataframe is stored in. Defaults to ``'obsl'``.
+
+        Returns:
+            Coordinate transform from the scene to the requested dataframe.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_transformation_to_multiscale_image(
```

Comment:
```suggestion
    def get_transformation_multiscale_image(
```

Do we need `to` here?

Reply:
We need something to specify what the transform is going from/to. (Scene to image is different than image to scene). I don't love how verbose the methods are, so I'm open to other suggestions. I did change `transformation` to match `transform` (based on one of your suggestion changes on #219).